app/src/main/java/com/group80/uoftinder/AcademicFilterActivity.java

```diff
@@ -0,0 +1,149 @@
+        programOfStudyBoxes = new CheckBox[] {
+                csBox, mpBox, lifeSciBox, socialSciBox, rotmanBox, otherStudyBox
+        };
+
+        CheckBox y1Box = findViewById(R.id.firstYearBox);
+        CheckBox y2Box = findViewById(R.id.secondYearBox);
+        CheckBox y3Box = findViewById(R.id.thirdYearBox);
+        CheckBox y4Box = findViewById(R.id.fourthYearBox);
+        CheckBox graduateBox = findViewById(R.id.graduateBox);
+        yearOfStudyBoxes = new CheckBox[]{y1Box, y2Box, y3Box, y4Box, graduateBox};
+
+        CheckBox stGeorgeBox = findViewById(R.id.stGeorgeBox);
+        CheckBox mississaugaBox = findViewById(R.id.mississaugaBox);
+        CheckBox scarboroughBox = findViewById(R.id.scarboroughBox);
+        campusBoxes = new CheckBox[]{stGeorgeBox, mississaugaBox, scarboroughBox};
+    }
+
+    /**
+     * Uncheck a box by toggling if the box is currently checked.
+     * @param checkBox  Android Checkbox object to uncheck
+     */
+    private void uncheckBox(CheckBox checkBox) {
+        if(checkBox.isChecked())
+            checkBox.toggle();
+    }
+
+    /**
+     * Return a set of integers representing user input to filter options
+     * represented by checkboxes.
+     * Integers included represent the index of the checked checkboxes.
+     * For example, {1, 2} represents the second and third checkboxes in
+     * checkboxes param were checked.
+     * @param checkboxes    A list of Android Checkbox objects
+     * @return              A set of integers containing the indices of
+     *                      the boxes in checkboxes that were checked.
+     */
+    private Set<Integer> populateCheckboxValues(CheckBox[] checkboxes) {
+        Set<Integer> checkboxValues = new HashSet<>();
+        for(int i = 0; i < checkboxes.length; i++) {
+            CheckBox checkBox = checkboxes[i];
+            if(checkBox.isChecked())
+                checkboxValues.add(i);
+        }
+        return checkboxValues;
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_academic_filter);
+
+        initializePickers();
+        initializeCheckBoxes();
+        Button filterButton = findViewById(R.id.filterButton);
+        Button resetButton = findViewById(R.id.resetButton);
+
+        filterButton.setOnClickListener(new View.OnClickListener() {
+            /**
+             * Returns to main page from filter view UI after updating the list
+             * of compatible users for the current user via RecommendationInteractor.
+             * @param view  Current view
+             */
+            @Override
+            public void onClick(View view) {
+                RecommendationInteractor recommendationInteractor = new RecommendationInteractor();
+                int minAge = minAgePicker.getValue();
+                int maxAge = maxAgePicker.getValue();
```

Comment:
Ah good catch! This thought did cross my mind at some point - just wasn't a priority in this PR. Definitely not a nitpick since it could very much happen to users - I think it would be the easiest for the View to catch this mismatch error via form validation. Just created a new issue #29 to reflect this as a next step.

Reply:
Sounds good!